### PR TITLE
fix: suppress warnings for different name representations of the same file in the case-insensitive file system when renaming

### PR DESCRIPTION
**Pull request by Xerxes-2**
Original issue here: https://github.com/sxyazi/yazi/issues/1171

Fixes https://github.com/sxyazi/yazi/issues/1171

This PR introduced a function `are_paths_equal` in `fns.rs` to determine if two paths points to the same node.
It fixes the rename and bulk_rename functions not working properly when only change the case of letters.
Previously these functions only check if the new path already exists, but not considering the old and new path could be the same node in case-insensitive systems like macOS and Windows.

### CODE REVIEW

File: yazi-shared/src/fs/fns.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use std::{borrow::Cow, collections::{HashMap, VecDeque}, ffi::{OsStr, OsString}, fs::Metadata, path::{Path, PathBuf}};
+use std::{borrow::Cow, collections::{HashMap, VecDeque}, ffi::{OsStr, OsString}, fs::Metadata, os::unix::fs::MetadataExt, path::{Path, PathBuf}};
```
Comment (lxl66566) [inline yazi-shared/src/fs/fns.rs:1]: unresolved import on windows, please move it to cfg(unix)

**Comment by lxl66566** [comment]
There's an unexpected behavior: when i rename a file to it's hardlink, it will overwrite (same as being deleted) without asking.
Before this PR, yazi would ask whether to Overwrite an existing file.

reproduce:
```
touch 1
ln 1 2
./yazi
(rename 2 to 1)
```

**Comment by Xerxes-2** [comment]
> There's an unexpected behavior: when i rename a file to it's hardlink, it will overwrite (same as being deleted) without asking. Before this PR, yazi would ask whether to Overwrite an existing file.
> 
> reproduce:
> 
> ```
> touch 1
> ln 1 2
> ./yazi
> (rename 2 to 1)
> ```

I think new commit have resolved this issue, but in a slight dumb way.

**Comment by sxyazi** [comment]
Thanks for the PR, I'll take care of the follow-up work on this.

**Review by sxyazi — APPROVED** [review]
Thanks!

**Pull request merged**